src/Workspaces/Core/Portable/Workspace/FileTextLoader.cs

```diff
@@ -58,10 +60,29 @@ public FileTextLoader(string path, Encoding? defaultEncoding)
 
         internal sealed override string FilePath => Path;
 
+        /// <summary>
+        /// Creates <see cref="SourceText"/> from <see cref="Stream"/>.
+        /// </summary>
+        /// <param name="stream">Stream.</param>
+        /// <param name="workspace">Obsolete. Dummy <see cref="Workspace"/> instance.</param>
+        [Obsolete("Use CreateText(Stream, CancellationToken)")]
         protected virtual SourceText CreateText(Stream stream, Workspace workspace)
+            => EncodedStringText.Create(stream, DefaultEncoding);
+
+        /// <summary>
+        /// Creates <see cref="SourceText"/> from <see cref="Stream"/>.
+        /// </summary>
+        protected virtual SourceText CreateText(Stream stream, CancellationToken cancellationToken)
+            => s_dummySourceText;
```

Comment:
Why not just have this call the other method w/o the CancellationToken being passed? Then you don't need the impl method doing something tricky?